scanpy/preprocessing/_simple.py

```diff
@@ -172,7 +175,7 @@ def filter_cells(
     if max_number is not None:
         cell_subset = number_per_cell <= max_number
 
-    s = materialize_as_ndarray(np.sum(~cell_subset))
+    s = axis_sum(~cell_subset)
     if s > 0:
```

Comment:
Why are we allowing this `if` statement to trigger `compute`?  I think this was a mistake unless we landed on some reason to do this and not just skip the logging.

Reply:
Is this because there is another `materialize_as_ndarray` call above (L145)? I don't think that should necessarily mean we should also do one here in effect.

Reply:
Leaving as-is because these functions need to do `compute` anyway due to subsetting ops